zio-schema-json/shared/src/main/scala/zio/schema/codec/JsonCodec.scala

```diff
@@ -62,14 +58,110 @@ object JsonCodec {
         JsonEncoder.charSequenceToByteChunk(jsonCodec.encodeJson(value, None))
 
       override def streamEncoder: ZPipeline[Any, Nothing, A, Byte] =
-        ZPipeline.mapChunks(
-          _.flatMap(encode)
-        )
+        ZPipeline.mapChunks[A, Chunk[Byte]](_.map(encode)).intersperse(Chunk.single('\n'.toByte)).flattenChunks
     }
 
   implicit def schemaBasedBinaryCodec[A](implicit schema: Schema[A]): BinaryCodec[A] =
     schemaBasedBinaryCodec[A](JsonCodec.Config.default)
 
+  val splitOnJsonBoundary: ZPipeline[Any, Nothing, String, String] = {
+    val validNumChars = Set('0', '1', '2', '3', '4', '5', '6', '7', '8', '9', 'E', 'e', '-', '+', '.')
+    ZPipeline.suspend {
+      val stringBuilder = new StringBuilder
+      var depth         = 0
+      var valueType     = 'j' // j = json, s = string, b = boolean, u = null, n = number, x = null after first null, e = escape in string
```

Comment:
Should we make val with this values? I don't thinkit would be a significant performance issue, but might be easier to understand than this

Reply:
I was inspired by `ZPipeline.splitLines` that also uses vars. The `valueType` is used for context to recognize how to interpret the current character and to spot when a section ends.

Reply:
Or did you mean using an enum / sealed trait?

Reply:
No just vals. Like a constants for the different states. Pretty sure a sealed trait is less performant

Reply:
Ok, I'll do that

Reply:
Done